src/expandableCalendar/WeekCalendar/presenter.js

```diff
@@ -0,0 +1,165 @@
+import _ from 'lodash';
+import React from 'react';
+import {sameWeek} from '../../dateutils';
+const commons = require('../commons');
+import XDate from 'xdate';
+import {toMarkingFormat} from '../../interface';
+
+const UPDATE_SOURCES = commons.UPDATE_SOURCES;
+// must be a positive number
+const NUMBER_OF_PAGES = 2;
+
+class Presenter {
+  constructor(props) {
+    this.list = React.createRef();
+    this.props = props;
```

Comment:
I don't see the need for `this.props` and I don't think it's a good practice. The presenter should get the props through the methods' parameters and not save them globally.

Reply:
Actually it's not in use, removed